### PR TITLE
Present parameter_entry as a record by consistency with other entries

### PR DESCRIPTION
**Pull request by herbelin**
For consistency of style, easier understanding of the code, and self-documentation of the code, we change `parameter_entry` from a tuple to a record.

I believe this is consensual enough to be proposed as a PR "in passing" while further works on coq/ceps#51, on #13445, on universes in sections are in preparation.

- [x] Overlay pull request: MetaCoq/metacoq#575

### CODE REVIEW

File: kernel/entries.ml [hunk 1]
```diff
@@ -40,7 +40,8 @@ type one_inductive_entry = {
   mind_entry_typename : Id.t;
   mind_entry_arity : constr;
   mind_entry_consnames : Id.t list;
-  mind_entry_lc : constr list }
+  mind_entry_lc : constr list;
+}
```
Comment (ppedrot) [inline kernel/entries.ml:44]: What about these unrelated cosmetic changes? (I prefer the new variant but it's a potential flamewar.)

Comment (herbelin) [reply kernel/entries.ml:44]: They are related in the sense of applying a `consistency of style`. I'm not saying that one is best than the other but in `entries.ml` at least, there is already a majority of single `}`. (Would it be in `declare.ml`, consistency of style may give instead a majority of `; field : type` style...)

Comment (ppedrot) [reply kernel/entries.ml:44]: As far as I am concerned, I hate the `; field : type` style with a passion. The semicolon is semantically related to the preceding field, not the rhs one.

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@herbelin it looks like metacoq needs an overlay.
